src/managers/search_manager.py

```
@@ -359,38 +383,134 @@ async def _debounced_search(self) -> None:
+
+        if not matched_paths:
+            return []
+
+        # マッチしたパスを優先度でソート(より具体的なパスを先に)
+        # 例: "profile.email" は "profile" より優先
+        matched_paths.sort(key=lambda x: (-x.count('.'), -x.count('['), x))
+
+        # 親パスを除外(子パスが存在する場合)
+        # 例: ["organization.name", "organization"] → ["organization.name"]
+        # セットベースの最適化: O(n)でプレフィックスセットを構築
+        path_set = set(matched_paths)
```

Ruff (0.14.6)

Local variable `path_set` is assigned to but never used

Remove assignment to unused variable `path_set`

(F841)